Kiss.Bff.EndToEndTest/NieuwsEnWerkInstructies.cs

```diff
@@ -1,180 +1,271 @@
 
-        var firstArticle = isBelangrijk
-            ? articles.First
-            : articles.Filter(new LocatorFilterOptions() { HasNot = Page.Locator(".featured") }).First;
+            // Create Message B and C with the current publish date
+            await CreateBericht("Message B: 724e44a3-6ba1-4e92-85c3-d44e35238f4a", false, "");
+            await CreateBericht("Important Message C: 5b8277a7-fb1a-4358-8099-24b9487b29bc", true, "");
 
-        var heading = firstArticle.GetByRole(AriaRole.Heading).First;
+            // Go to the page and retrieve the order of articles
+            await Page.GotoAsync("/");
+            await Page.WaitForSelectorAsync("article");
+            var allArticles = NieuwsSection.GetByRole(AriaRole.Article);
 
-        await Expect(heading).ToHaveTextAsync(title);
-        await MarkAllNewsItems(false);
-    }
+            // Dictionary to hold article positions
+            var initialOrderOnPage = new Dictionary<string, int>();
 
-    [TestMethod]
-    public async Task Als_ik_een_belangrijk_bericht_publiceer_komt_deze_bovenaan()
-    {
-        var titel = $"End to end test {Guid.NewGuid()}";
-        var featuredIndicator = Page.Locator(".featured-indicator");
-        var intitialFeatureCount = await featuredIndicator.IsVisibleAsync()
-            && int.TryParse(await featuredIndicator.TextContentAsync(), out var c)
-                ? c
-                : 0;
-
-        await CreateBericht(titel, true);
-        
-        try
-        {
+            for (var index = 0; index < await allArticles.CountAsync(); index++)
+            {
+                var element = allArticles.Nth(index);
+                var innerHtml = await element.InnerTextAsync();
+
+                if (innerHtml.Contains("Message A: 8e600d44-81fb-4302-9675-31b687619026"))
+                {
+                    initialOrderOnPage.Add("Message A", index);
+                }
+                if (innerHtml.Contains("Message B: 724e44a3-6ba1-4e92-85c3-d44e35238f4a"))
+                {
+                    initialOrderOnPage.Add("Message B", index);
+                }
+                if (innerHtml.Contains("Message C: 5b8277a7-fb1a-4358-8099-24b9487b29bc"))
+                {
+                    initialOrderOnPage.Add("Message C", index);
+                }
+            }
+
+            var indexVanA = initialOrderOnPage["Message A"];
+            var indexVanB = initialOrderOnPage["Message B"];
+            var indexVanC = initialOrderOnPage["Message C"];
+
+            Assert.IsTrue(indexVanC < indexVanB && indexVanB < indexVanA, "Initial order should be C, B, A.");
+
+            // Act: Update message A
+            await UpdateBericht("Message A: 8e600d44-81fb-4302-9675-31b687619026", "Updated Message A: 8e600d44-81fb-4302-9675-31b687619026");
+
+            // Refresh page and retrieve articles again
             await Page.GotoAsync("/");
+            await Page.WaitForSelectorAsync("article");
```

Comment:
dit is de fix? wat doet dit? wacht het tot er een willekeurig article element op de pagina staat?

Reply:
jep, hij wacht op een article en gaat daarna pas verder de test uitvoeren

Reply:
maar het is toch niet robuust om op een volledig willekeurige article tag ergens op het scherm te wachten? moet je er niet een wat specifiekere selector van maken die op nieuwbericht articles wacht?

Reply:
ik heb er nu specifieke selectors van gemaakt